indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -49,7 +57,166 @@
 {
   private static final Logger log = new Logger(InputRowSerde.class);
 
-  public static final byte[] toBytes(final InputRow row, AggregatorFactory[] aggs, boolean reportParseExceptions)
+  private static final IndexSerdeTypeHelper STRING_HELPER = new StringIndexSerdeTypeHelper();
+  private static final IndexSerdeTypeHelper LONG_HELPER = new LongIndexSerdeTypeHelper();
+  private static final IndexSerdeTypeHelper FLOAT_HELPER = new FloatIndexSerdeTypeHelper();
+  private static final IndexSerdeTypeHelper DOUBLE_HELPER = new DoubleIndexSerdeTypeHelper();
+
+  public interface IndexSerdeTypeHelper<T>
+  {
+    ValueType getType();
+
+    byte[] serialize(Object value);
+
+    T deserialize(byte[] bytes);
+  }
+
+  private static final IndexSerdeTypeHelper[] VALUE_TYPE_HELPER_ARRAY =
+      new IndexSerdeTypeHelper[ValueType.values().length];
+  static {
+    VALUE_TYPE_HELPER_ARRAY[ValueType.STRING.ordinal()] = STRING_HELPER;
+    VALUE_TYPE_HELPER_ARRAY[ValueType.LONG.ordinal()] = LONG_HELPER;
+    VALUE_TYPE_HELPER_ARRAY[ValueType.FLOAT.ordinal()] = FLOAT_HELPER;
+    VALUE_TYPE_HELPER_ARRAY[ValueType.DOUBLE.ordinal()] = DOUBLE_HELPER;
+  }
+
+  public static Map<String, IndexSerdeTypeHelper> getTypeHelperMap(DimensionsSpec dimensionsSpec)
+  {
+    Map<String, IndexSerdeTypeHelper> typeHelperMap = Maps.newHashMap();
+    for (DimensionSchema dimensionSchema : dimensionsSpec.getDimensions()) {
+      IndexSerdeTypeHelper typeHelper;
+      switch (dimensionSchema.getValueType()) {
```

Comment:
Replace the switch with `VALUE_TYPE_HELPER_ARRAY[dimensionSchema.getValueType().ordinal()]` ?

Reply:
I kept the switch statement but dropped the array, it wasn't being used anymore after removing the type ordinals from the serialized form